modules/ossm-extensions-migrating-to-wasmplugin.adoc

```diff
@@ -0,0 +1,34 @@
+////
+This module included in the following assemblies:
+*service_mesh_/v2x/ossm-extensions.adoc
+////
+:_content-type: PROCEDURE
+[id="ossm-extensions-migrating-to-wasmplugin_{context}"]
+= Migrating to `WasmPlugin` resources
+
+To upgrade your WebAssembly extensions from the `ServiceMeshExtension` API to the `WasmPlugin` API, you remove the old plugin and create a new plugin file. Creating an extension file involves replacing the extension and compiling it into a WebAssembly module, and then re-deploying it into a container.
```

Comment:
Strictly speaking, the plugin format itself has not changed. It still is a `.wasm` file. Just to recap: A `.wasm` file is the result of compiling a source code (written in c++, go, rust, etc) into the WASM binary format. The only requirement for the new Istio's `WasmPlugin` API is that this file is named `plugin.wasm`. It's not necessary to change the source code or even recompile it to produce a new `.wasm` file. Just renaming it (if it's not already) to `plugin.wasm` is necessary.

The `ServiceMeshExtension` and `WasmPlugin` API's are just two different ways to make the bridge between Istio and Envoy. Let's say one is a concrete bridge and another one is a wood bridge. The car is the `.wasm` file that needs to cross the bridge. The car is the same, it doesn't need to change. We can see this in the both snippets above. Both API's consume the same container image (which contains the same `.wasm` file).

That said, the statement above is confusing to me. I think it could be entirely removed without loss.

Reply:
>A .wasm file is the result of compiling a source code (written in c++, go, rust, etc) into the WASM binary format.

Thank you, that's a very clear explanation.  I'm so used to thinking that we do everything via YAML files in service mesh, I forgot that we're talking about code here.